src/og/entity/GeoObjectHandler.js

```diff
@@ -0,0 +1,1106 @@
+            });
+        } else {
+            const prevState = this._instancedTags.get(tag),
+                nextCount = prevState.iCounts + 1;
+            this._instancedTags.set(tag, {
+                vCounts: geoObject._verticesCount * itemSize,
+                iCounts: nextCount,
+                iSize: prevState.iSize,
+                index: prevState.index
+            });
+        }
+        const tagData = this._instancedTags.get(tag),
+            ti = tagData.index;
+
+        /**
+         * mark object by index for recalc indices array
+         */
+        geoObject._tagIndex = tagData.iCounts - 1;
+        if (geoObject._visibility) {
+            if (!alreadyAdded) {
+                this._vertexArr[ti] = concatArrays(
+                    this._vertexArr[ti],
+                    setParametersToArray(
+                        [],
+                        0,
+                        geoObject._verticesCount * itemSize,
+                        geoObject._verticesCount * itemSize,
+                        ...geoObject._vertices
+                    )
+                );
+                this._normalsArr[ti] = concatArrays(
+                    this._normalsArr[ti],
+                    setParametersToArray(
+                        [],
+                        0,
+                        geoObject._verticesCount * itemSize,
+                        geoObject._verticesCount * itemSize,
+                        ...geoObject._normals
+                        // ...getTriangleNormals(geoObject._vertices, geoObject._indices)
+                    )
+                );
+            }
+        } else {
+            this._vertexArr[ti] = concatArrays(
+                this._vertexArr[ti],
+                setParametersToArray([], 0, geoObject._verticesCount * itemSize, 1, 0)
+            );
+        }
+        var x = geoObject._positionHigh.x,
+            y = geoObject._positionHigh.y,
+            z = geoObject._positionHigh.z,
+            w;
+
+        this._positionHighArr[ti] = concatArrays(
+            this._positionHighArr[ti],
+            setParametersToArray([], 0, itemSize, itemSize, x, y, z)
+        );
+        x = geoObject._positionLow.x;
+        y = geoObject._positionLow.y;
+        z = geoObject._positionLow.z;
+        this._positionLowArr[ti] = concatArrays(
+            this._positionLowArr[ti],
+            setParametersToArray([], 0, itemSize, itemSize, x, y, z)
+        );
+
+        x = geoObject._entity._pickingColor.x / 255;
+        y = geoObject._entity._pickingColor.y / 255;
+        z = geoObject._entity._pickingColor.z / 255;
+        this._pickingColorArr[ti] = concatArrays(
+            this._pickingColorArr[ti],
+            setParametersToArray([], 0, itemSize, itemSize, x, y, z)
+        );
+
+        x = geoObject._direction.x;
+        y = geoObject._direction.y;
+        z = geoObject._direction.z;
+        this._directionArr[ti] = concatArrays(
```

Comment:
In debugger NaN values in the array when an object is appeared. Please, check geoObject._direction object

Reply:
Ok, I'm check it

Reply:
It's happen in function `updateDirection`
`Planet.getBearingNorthRotationQuat(this._position)` on start return zero vector.
Next `mul` in quaternion return zero quaternion. And, finally normalize of mul vector and zero quaternion return vector with `NaN`. Normalize in `Vec3` don't check by 0 before divide

Reply:
fixed